src/timeseriesinsights/azext_timeseriesinsights/_params.py

```diff
@@ -0,0 +1,88 @@
+        c.argument('storage_account_name', arg_group="Storage Configuration", help='The name of the storage account that will hold the environment\'s long term data.')
+        c.argument('storage_management_key', arg_group="Storage Configuration", help='The value of the management key that grants the Time Series Insights service write access to the storage account. This property is not shown in environment responses.')
+        c.argument('time_series_id_properties', nargs='+')
+        c.argument('data_retention', type=iso_8601_timespace, help='The number of days the environment\'s events will be available for query from the warm store.')
+    # endregion
+
+    # region event-source
+    with self.argument_context('timeseriesinsights event-source') as c:
+        from .vendored_sdks.timeseriesinsights.models import LocalTimestampFormat
+        c.argument('environment_name', arg_type=environment_name_type)
+        c.argument('event_source_name', arg_type=name_type, id_part='child_name_1', help='The name of the Time Series Insights event source associated with the specified environment.')
+        c.argument('local_timestamp_format', arg_group="Local Timestamp", arg_type=get_enum_type(LocalTimestampFormat), help='An enum that represents the format of the local timestamp property that needs to be set. Currently only Embedded is supported.')
+        c.ignore('time_zone_offset_property_name')
+        # c.argument('time_zone_offset_property_name', arg_group="Local Timestamp", help='The event property that will be contain the offset information to calculate the local timestamp. When the LocalTimestampFormat is Iana, the property name will contain the name of the column which contains IANA Timezone Name (eg: Americas/Los Angeles). When LocalTimestampFormat is Timespan, it contains the name of property which contains values representing the offset (eg: P1D or 1.00:00:00)')
+    # endregion
+
+    # region reference-data-set
+    with self.argument_context('timeseriesinsights reference-data-set') as c:
+        c.argument('environment_name', arg_type=environment_name_type)
+        c.argument('reference_data_set_name', arg_type=name_type, id_part='child_name_1', help='Name of the reference data set.')
+
+    with self.argument_context('timeseriesinsights reference-data-set create') as c:
+        from .vendored_sdks.timeseriesinsights.models import DataStringComparisonBehavior
+        c.argument('key_properties', nargs='+', help='The list of key properties for the reference data set. Format: NAME TYPE ...')
+        c.argument('data_string_comparison_behavior', arg_type=get_enum_type(DataStringComparisonBehavior))
+    # endregion
+
+    # region access-policy
+    with self.argument_context('timeseriesinsights access-policy') as c:
+        from .vendored_sdks.timeseriesinsights.models import AccessPolicyRole
+        c.argument('resource_group', resource_group_name_type)
```

Comment:
>resource_group_name_type [](start = 37, length = 24)

Unnecessary to define `resource_group` here as it's a global argument added by cli core automatically.

Reply:
✔